inventory/byo/hosts.origin.example

```diff
@@ -89,6 +89,21 @@ openshift_release=v1.4
 # Skip upgrading Docker during an OpenShift upgrade, leaves the current Docker version alone.
 # docker_upgrade=False
 
+
+# Upgrade Hooks
+#
+# Hooks are available to run custom tasks at various points during a cluster
+# upgrade. Each hook should point to a file with Ansible tasks defined. Suggest using
+# absolute paths, if not the path will be treated as relative to the file where the
+# hook is actually used.
+#
+# Tasks to run before each master is upgraded:
+# openshift_upgrade_pre_master_hook=/usr/share/custom/pre_master.yml
```

Comment:
Good info thanks @wester55 

I think a few of these have been improved in 3.4 and more in 3.5 but boiling this down a bit:

- fix bug where schedulable masters are left unschedulable
- upgrade masters serially (just committed for 3.5, previously we did a bunch in parallel and then restarted services serially)
- implement hooks before and after each master is upgraded (this PR)
- similar hooks for each node

Additionally we plan to make master/control plane upgrade *also* handle the node/docker on the masters, and then omit these later when we're just upgrading nodes. This will make the process of upgrading a master more atomic.